Mithril update 2024-04-24 (#383)
IntersectMBO/cardano-updates: blog/2024-04-24-mithril.md

---
title: Mithril Team Update
slug: 2024-04-24-mithril
authors: jpraynaud
tags: [mithril]
hide_table_of_contents: false
---

## High level overview

> This week, most of the Mithril team attended the [Cardano Buidler Fest #1](https://buidl.2024.cardano.org/) in Toulouse, France. They also continued implementing the certification of Cardano transactions in Mithril networks. They kept working on scaling the signature and proof generation for `mainnet` with the compression of the transaction Merkle tree by using sub-Merkle trees of transactions by block ranges. They identified the origin of the issue preventing the memory from being properly released when signing/proving large set of transactions and worked on a fix for it. Additionally, they completed the prototype for decentralizing signer registration with the relay and a peer-to-peer (P2P) network.
> 
> Finally, the team completed the implementation of a configurable feature for test networks to log unparsable blocks instead of panicking and investigated some unexpected error logs occurring on the Cardano node when signer and aggregator connect to the mini-protocols.

Reviewer: ```suggestion
This week, most of the Mithril team attended the [Cardano Buidler Fest #1](https://buidl.2024.cardano.org/) in Toulouse, France. They also continued implementing the certification of Cardano transactions in Mithril networks and worked on scaling the signature and proof generation for `mainnet` by compressing the transaction Merkle tree using sub-Merkle trees based on transaction block ranges. The team identified the source of an issue preventing proper memory release during the signing/proving of a large set of transactions and developed a fix for it. Additionally, they completed the prototype for decentralizing signer registration with the relay and a peer-to-peer (P2P) network.

Finally, the team implemented a configurable feature for test networks to log unparsable blocks instead of panicking and investigated some unexpected error logs occurring on the Cardano node when the signer and aggregator connect to the mini-protocols.
```
